src/commands.rs

```diff
@@ -2124,13 +2095,12 @@ fn cmd_checkout(
     args: &CheckoutArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let new_commit = workspace_command.resolve_single_rev(ui, &args.revision)?;
+    let new_commit = workspace_command.resolve_single_rev(&args.revision)?;
     let workspace_id = workspace_command.workspace_id();
     if ui.settings().enable_open_commits() {
         if workspace_command.repo().view().get_checkout(&workspace_id) == Some(new_commit.id()) {
             ui.write("Already on that commit\n")?;
         } else {
-            workspace_command.commit_working_copy(ui)?;
```

Comment:
Apparently `jj checkout --no-commit-working-copy` no longer fails.
It failed at `commit_working_copy()` before.

Maybe we need to insert `check_working_copy_writable()` to more places?

Reply:
I actually made that change on purpose. It's not technically necessary to update the working copy in order to update the recorded checkout in the repo view. Perhaps we should instead add a general check to prevent mutations when the user passes `--no-commit-working-copy` or `--at-op=`, unless they also pass `--allow-divergent-operations` or something like that. I think I'll work on a patch for that if you agree.

(I'm slow to respond because I'm on vacation.)

Reply:
> I actually made that change on purpose. It's not technically necessary to update the working copy

Okay, I just thought it would be a oversight. If that's on purpose, I'm not against the new behavior.

> add a general check to prevent mutations when the user passes --no-commit-working-copy or --at-op=, unless they also pass --allow-divergent-operations

If `jj checkout --no-commit-working-copy` leads to scary state or future data loss, `--allow-divergent-operations` would be useful. If it's easy to recover from the divergence, I think no more protection will be needed.

Reply:
> > I actually made that change on purpose. It's not technically necessary to update the working copy
> 
> Okay, I just thought it would be a oversight. If that's on purpose, I'm not against the new behavior.

Yes, sorry I was too lazy to document it in the commit message. I've updated it now.

> 
> > add a general check to prevent mutations when the user passes --no-commit-working-copy or --at-op=, unless they also pass --allow-divergent-operations
> 
> If `jj checkout --no-commit-working-copy` leads to scary state or future data loss, `--allow-divergent-operations` would be useful. If it's easy to recover from the divergence, I think no more protection will be needed.

Divergent operations are currently resolved by arbitrarily picking the checkout from one side. There should be no data loss, only possible confusion. It should be possible to recovery by `jj undo` or `jj op restore`. So we can consider adding a check for divergent operations later.